bot_modules/irc_commands.js

```diff
@@ -482,6 +482,67 @@ module.exports = {
 			})
 		})
 	},
+  
+	/**
+	 *  roll
+	 *
+	 */
+	roll: (info, words) => {
+		let dice_notation = words.slice(1).join(" ").toLowerCase();
+		if (dice_notation.includes("d")) {
+	   		if (dice_notation == "") dice_notation = "1d10";
+	        
+	   		if (dice_notation.startsWith("d")) {dice_notation = "1" + dice_notation; var dice_amount = 1}
+	   		else {var dice_amount = parseInt(dice_notation.split("d")[0])};
+	        
+	   		let sum = 0;
+	        
+	   		var dice_amount_length = dice_amount.toString().length + 1;
+	        
+	  		if (dice_notation.includes("+")) {
+	    			let add_to_sum = dice_notation.split("+")[1];
+	            
+	    			if (add_to_sum.includes("d")) {
+	     				if (add_to_sum.startsWith("d")) add_to_sum = "1" + add_to_sum;
+	     				let subdice_amount = parseInt(add_to_sum.split("d")[0]);;
+	     				let subnumber_limit = parseInt(add_to_sum.split("d")[1]);
```

Comment:
what happens when users pass giant long numerical strings?
when we had this feature up briefly someone crashed it with a very long string

Reply:
because of ``parseInt()``, I think, I will see if there is another way of converting to numbers

Reply:
maybe just truncate the numerical string first 
not sure whats best but could enforce some kind of max value 🤷

Reply:
large numbers seem to work: 
<viraxor> !roll 238912084723
<BotB2_tes>  viraxor rolls 152164191151! 
<viraxor> !roll 4327849327485734957234875698237423784897213985484
<BotB2_tes>  viraxor rolls 3.6274747582713556e+48!
though this exponent is a small problem. js does that automatically, but it's easy to patch